proposals/2024-09-25_metadata-labels.md

```diff
@@ -0,0 +1,216 @@
+* All API parts (Series, LabelNames, LabelValues, Recordings/Alerts, remote APIs) will expose new labels without control. This means ecosystem will start depending on this, once this feature gets more mature, **potentially prohibiting the alternative approaches (e.g. only exposing `~seconds.counter` special syntax instead of raw `__type__=~".*"` selectors)**. We accept that risk.
+* Downstream users might be surprised by the new labels e.g. in Cortex, Thanos, Mimir, vendors.
+
+### Open Questions
+
+1. Should we prohibit standalone type and unit selectors for performance reasons? This would be inconsistent with __name__ though?
+
+### Implementation Plan
+
+#### Milestone 1: Feature flag for `type-and-unit-labels`
+
+See: https://github.com/prometheus/prometheus/pull/16228
+
+Add a feature flag: `--enable-feature=type-and-unit-labels`. When enabled, `__type__` and `__unit__` labels are added when receiving OTLP or PRW 2.0, or when scraping the text, OM, or proto formats. Implement any changes required to allow relabeling `__type__` and `__unit__` in `metric_relabel_configs`. 
+
+This will also handle basic PromQL semantics (when __name__ is dropped we also drop __type__ and __unit__)
+
+#### Milestone 2: UI and further PromQL changes
+
+During this stage, implement the UI changes. Iterate based on feedback. Changes should have no effect unless the `type-and-unit-labels` feature flag is enabled.
+
+#### Milestone 3: Add no translation option for OTLP translation
+
+Add an option for the OTLP no translation strategy. When enabled, it disables UTF-8 sanitization and the addition of suffixes. In the documentation for this option, recommend that `--enable-feature=type-and-unit-labels` is enabled.
+
+## Alternatives
+
+### Type and Unit in complex value types
+
+Unlike other metric types, native histograms are a complex type and contain many fields in a single value. The metric type and unit could be added as fields of a complex value type in a similar way.
+
+This solution is not chosen because:
+
+* Requires intrusive changes to all formats (text, proto, etc.).
+* Requires new PromQL syntax for querying the type and unit.
+
+### "Hide" __type__ and __unit__ labels in PromQL, instead of UI
+
+Existing UIs don't handle the `__type__` and `__unit__` labels. To mitigate this, PromQL could omit the `__type__` and `__unit__` labels from the query response. Doing this would avoid requiring UIs to update to handle the new labels.
+
+This solution is not chosen because:
+
+* It deviates from the current handling of the `__name__` label.
+* We expect metadata, like type and unit to be useful to display in the UI, and want to enable these use-cases.
+* It should be a small amount of effort to hide these labels.
+
+### Omit __unit__ label from counting series for summaries and histograms
```

Comment:
I'm a bit confused by this section - I'm unsure what the chosen proposal is vs the rejected alternatives, mostly because I don't see a histogram unit proposal within the How section to correlate with this. My assumption is that the proposal is "The unit of a `_count` or `_bucket` series of a histogram is always `1`" and the rest is the alternatives - is that correct?

Reply:
Made this more clear.